pylib/gyp/__init__.py

```diff
@@ -103,6 +103,18 @@ def Load(
     for (key, val) in generator.generator_default_variables.items():
         default_variables.setdefault(key, val)
 
+    output_dir = params["options"].generator_output or params["options"].toplevel_dir
+    if default_variables["GENERATOR"] == "ninja":
+        default_variables.setdefault(
+            "PRODUCT_DIR_ABS",
+            os.path.join(output_dir, "out", default_variables["build_type"]),
+        )
+    else:
+        default_variables.setdefault(
+            "PRODUCT_DIR_ABS",
+            os.path.join(output_dir, default_variables["CONFIGURATION_NAME"]),
```

Comment:
Was there a reason this is different from https://github.com/nodejs/node/commit/f4bd91b0e2d7f07365ccf6bba943ba97d6f95cc9#diff-b444d542a5859a68304f84ed3e5a6c81730671295236c26c7866eab5306c1d11 where both ninja and non-ninja paths used `default_variables["build_type"]`?

Reply:
See https://github.com/nodejs/node/pull/42986.